apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -25,17 +33,27 @@ const getById = (id: number) =>
   db.selectFrom('Resource').where('Resource.id', '=', id)
 
 // NOTE: Throw here to fail early if our invariant that a page has a `blobId` is violated
-export const getFullPageById = (id: number) => {
+export const getFullPageById = async (id: number) => {
+  // Check if draft blob exists and return that preferentially
+  const draftBlob = await getById(id)
+    .where('Resource.draftBlobId', 'is not', null)
+    .innerJoin('Blob', 'Resource.draftBlobId', 'Blob.id')
+    .selectAll()
```

Comment:
we should not return everything back to the frontend in case we return more data than necessary. should we select specific columns that the frontend needs to return?

can even extract into something like `defaultDraftSelect`

Reply:
extracted, currently is all the columns but we can change this later